core/imgread/common.cpp

```diff
@@ -138,158 +152,122 @@ Disc* OpenDisc(const wchar* fn)
+	settings.imgread.LastImage = diskImageFileName;
 
 	SaveSettings();
 
-	if (!InitDrive_(fn))
-	{
-		//msgboxf("Selected image failed to load",MBX_ICONERROR);
-			NullDriveDiscType=NoDisk;
-			gd_setdisc();
-			sns_asc=0x29;
-			sns_ascq=0x00;
-			sns_key=0x6;
-		return true;
-	}
-	else
+	if (!InitDrive_(diskImageFileName.c_str()))
 	{
-		return true;
+		NullDriveDiscType = NoDisk;
+		gd_setdisc();
+		SetSnsCodes(NoDiskInsertedSenseCode);
+		return false;
 	}
+	return true;
 }
 
 bool DiscSwap(u32 fileflags)
 {
-	if (settings.imgread.LoadDefaultImage)
-	{
-		printf("Loading default image \"%s\"\n",settings.imgread.DefaultImage);
-		if (!InitDrive_(settings.imgread.DefaultImage))
-		{
-			msgboxf("Default image \"%s\" failed to load",MBX_ICONERROR);
-			return false;
-		}
-		else
-			return true;
-	}
-
-	// FIXME: Data loss if buffer is too small
-	wchar fn[512];
-	strncpy(fn, settings.imgread.LastImage, sizeof(fn));
-	fn[sizeof(fn) - 1] = '\0';
-
+    if (settings.imgread.LoadDefaultImage)
+    {
+        printf("Loading default image \"%s\"\n", settings.imgread.DefaultImage.c_str());
+        if (!InitDrive_(settings.imgread.DefaultImage.c_str()))
+        {
+            msgboxf("Default image \"%s\" failed to load", MBX_ICONERROR);
+            return false;
+        }
+        else
+            return true;
+    }
+
+    std::string diskImageFileName{ settings.imgread.LastImage };
 
 #ifdef BUILD_DREAMCAST
-	int gfrv=GetFile(fn,0,fileflags);
+    int gfrv = GetFile(diskImageFileName);
 #else
-	int gfrv=0;
+    int gfrv = rv_error;
 #endif
-	if (gfrv == 0)
-	{
-		NullDriveDiscType=Open;
-		gd_setdisc();
-		sns_asc=0x28;
-		sns_ascq=0x00;
-		sns_key=0x6;
-		return true;
-	}
-	else if (gfrv == -1)
-	{
-		sns_asc=0x28;
-		sns_ascq=0x00;
-		sns_key=0x6;
-		return false;
-	}
-
-	// FIXME: Data loss if buffer is too small
-	strncpy(settings.imgread.LastImage, fn, sizeof(settings.imgread.LastImage));
-	settings.imgread.LastImage[sizeof(settings.imgread.LastImage) - 1] = '\0';
-
-
-	SaveSettings();
-
-	if (!InitDrive_(fn))
-	{
-		//msgboxf("Selected image failed to load",MBX_ICONERROR);
-		NullDriveDiscType=Open;
-		gd_setdisc();
-		sns_asc=0x28;
-		sns_ascq=0x00;
-		sns_key=0x6;
-		return true;
-	}
-	else
-	{
-		sns_asc=0x28;
-		sns_ascq=0x00;
-		sns_key=0x6;
-		return true;
-	}
+    if (gfrv == rv_ok)
+    {
+        NullDriveDiscType = Open;
+        gd_setdisc();
+        SetSnsCodes(NoDiskErrorSenseCode);
+        return true;
+    }
+    else
+    {
+        SetSnsCodes(NoDiskInsertedSenseCode);
+        return false;
+    }
+
+
+    settings.imgread.LastImage = diskImageFileName;
+
+    SaveSettings();
+
+    if (!InitDrive_(diskImageFileName.c_str()))
+    {
+        msgboxf("Selected image failed to load", MBX_ICONERROR);
+        NullDriveDiscType = Open;
+        gd_setdisc();
+        SetSnsCodes(NoDiskInsertedSenseCode);
+        return false;
+    }
+    else
+    {
+        SetSnsCodes(NoDiskErrorSenseCode);
```

Comment:
Is this really correct here?

Reply:
I suggest I poorly chose a variable name. NoDiskErrorSenseCode - is that a "NoDisc!" error, or it is a "No Disc Error, everything's fine!".